ShopCartMap.java

```diff
@@ -0,0 +1,102 @@
+package com.baluybs.shop;
+import java.util.*;
+
+class ProductMap{
+	private String itemName;
+	private int itemPrice;
+	private int quantity;
+	public String getItemName() {
+		return itemName;
+	}
+	public void setItemName(String itemName) {
+		this.itemName = itemName;
+	}
+	public int getItemPrice() {
+		return itemPrice;
+	}
+	public void setItemPrice(int itemPrice) {
+		this.itemPrice = itemPrice;
+	}
+	public int getQuantity() {
+		return quantity;
+	}
+	public void setQuantity(int quantity) {
+		this.quantity = quantity;
+	}
+	@Override
+	public String toString() {
+		return itemName + itemPrice + quantity+"\n";
+	}
+}
+public class ShopCartMap {
+	static Map<Integer, ProductMap> m=new HashMap<>();
+	ProductMap p=null;
+	double totalPrice=0.0;
+	public void showItems() {
+		System.out.println("Pid    Products    Price(rs)");
+		System.out.println("1      kurkure     10");
+		System.out.println("2      Lays        20");
+		System.out.println("3      Oreo        40");
+	}
+	public void addProduct(int pid,int qnty) {
+		if (pid==1) {
+			if (m.containsKey(pid)) {
+				ProductMap k=(ProductMap)m.get(pid);
+				k.setQuantity(qnty+(int)k.getQuantity());
+				m.put(pid, k);
+			}else {
+			p=new ProductMap();
+			p.setItemName("kurkure");
+			p.setItemPrice(10);
+			p.setQuantity(qnty);
+			m.put(1, p);
+			}
+		}
+		else if (pid==2) {
+			if (m.containsKey(pid)) {
+				ProductMap k=(ProductMap)m.get(pid);
+				k.setQuantity(qnty+(int)k.getQuantity());
+				m.put(pid, k);
+			}else {
+			p=new ProductMap();
+			p.setItemName("Lays");
+			p.setItemPrice(20);
+			p.setQuantity(qnty);
+			m.put(2, p);
+			}
+		}
+		else if (pid==3) {
+			if (m.containsKey(pid)) {
+				ProductMap k=(ProductMap)m.get(pid);
+				k.setQuantity(qnty+(int)k.getQuantity());
+				m.put(pid, k);
+			}else {
+			p=new ProductMap();
+			p.setItemName("Oreo");
+			p.setItemPrice(40);
+			p.setQuantity(qnty);
+			m.put(3, p);
+			}
```

Comment:
Very bad approach 
change it because you are doing the same work in the different if else parts
you can create a array and set product price in that and name(in different array) as well.

Reply:
Actually I am using map so I don't want to use array and I didn't get how to use array in that I use map to store everything or can I write update method separately and can I add that process?